File: libs/language/src/lib/mapping/__tests__/schema/personListExample.ts
import z from 'zod';
import {
  blueDescriptionField,
  blueNameField,
  blueNodeField,
  withTypeBlueId,
} from '../../annotations';
import { personSchema } from './person';

export const personListExampleSchema = withTypeBlueId('PersonList-BlueId')(
  z.object({
    team1: z.array(personSchema),
    team2Name: blueNameField('team2'),
    team2Description: blueDescriptionField('team2'),
    team2: z.array(personSchema),
    team3: blueNodeField(),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/PersonListExample.java
package blue.language.mapping.model;

import blue.language.model.*;

import java.util.List;

@TypeBlueId("PersonList-BlueId")
public class PersonListExample {
    public List<? extends Person> team1;
    @BlueName("team2")
    public String team2Name;
    @BlueDescription("team2")
    public String team2Description;
    public List<? extends Person> team2;
//    public PersonList team3;
//    public Map<String, Object> team4;
    public Node team3;
    /**
     * team:
     *   - type:
     *       blueId: Doctor-BlueId
     *     name: Adam
     *     specialization: surgeon
     *   - type:
     *       blueId: Nurse-BlueId
     *     name: Betty
     *     yearsOfExperience: 12
     */
    public static class PersonList {
        public String name;
        public String description;
        public List<? extends Person> value;
    }
}
